### ISSUE TITLE
ruby193-jekyll seems to depend on wrong directory_watcher

### ISSUE THREAD

**Issue by davepacheco** [TritonDataCenter/pkgsrc-legacy#241]
I'm using pkgsrc on OS X using:

```
$ cat /usr/pkg/etc/pkg_install.conf 
PKG_PATH=http://pkgsrc.smartos.org/packages/Darwin/2013Q4/i386/All
```

I installed ruby193-jekyll:

```
$ sudo pkgin -y in ruby193-jekyll
calculating dependencies... done.

nothing to upgrade.
30 packages to be installed: libxcb-1.9.3 libXdmcp-1.1.1 libXau-1.0.8 libXext-1.3.2 libX11-1.6.2 libffi-3.0.13 gdbm-1.10nb3 tk-8.5.15 tcl-8.5.15 ruby193-tk-1.9.3p484nb1 ruby193-readline-1.9.3p484 ruby193-gdbm-1.9.3p484 ruby193-fiddle-1.9.3p484 ruby193-curses-1.9.3p484nb1 libyaml-0.1.4 ruby193-fast-stemmer-1.0.2 ruby193-highline-1.6.19 ruby193-syntax-1.0.0nb1 ruby193-safe_yaml-0.9.5 ruby193-redcarpet-3.0.0 ruby193-maruku-0.6.1 ruby193-liquid-2.5.2 ruby193-kramdown-0.14.2 ruby193-directory-watcher-1.5.1 ruby193-commander-4.1.5 ruby193-colorator-0.1 ruby193-classifier-1.3.3 ruby193-base-1.9.3p484 ruby193-pygments.rb-0.5.2 ruby193-jekyll-1.2.0 (23M to download, 92M to install)

downloading packages...
libxcb-1.9.3.tgz                         100%  661KB  94.4KB/s  46.9KB/s   00:07    
libXdmcp-1.1.1.tgz                       100%   31KB  31.3KB/s  31.3KB/s   00:00    
libXau-1.0.8.tgz                         100%   12KB  11.7KB/s  11.7KB/s   00:00    
libXext-1.3.2.tgz                        100%  107KB  53.5KB/s  61.1KB/s   00:02    
libX11-1.6.2.tgz                         100% 1997KB 285.3KB/s 171.9KB/s   00:07    
libffi-3.0.13.tgz                        100%   37KB  37.2KB/s  37.2KB/s   00:01    
gdbm-1.10nb3.tgz                         100%   93KB  92.7KB/s  92.7KB/s   00:01    
tk-8.5.15.tgz                            100% 2079KB 189.0KB/s  66.1KB/s   00:11    
tcl-8.5.15.tgz                           100% 2269KB 174.5KB/s  99.7KB/s   00:13    
ruby193-tk-1.9.3p484nb1.tgz              100% 1681KB 240.2KB/s 136.1KB/s   00:07    
ruby193-readline-1.9.3p484.tgz           100%   13KB  13.5KB/s  13.5KB/s   00:00    
ruby193-gdbm-1.9.3p484.tgz               100%   11KB  10.8KB/s  10.8KB/s   00:00    
ruby193-fiddle-1.9.3p484.tgz             100% 9494     9.3KB/s   9.3KB/s   00:00    
ruby193-curses-1.9.3p484nb1.tgz          100%   20KB  19.5KB/s  19.5KB/s   00:00    
libyaml-0.1.4.tgz                        100%  101KB  50.7KB/s  57.3KB/s   00:02    
ruby193-fast-stemmer-1.0.2.tgz           100%   72KB  72.1KB/s  23.1KB/s   00:01    
ruby193-highline-1.6.19.tgz              100%  552KB 276.1KB/s  19.2KB/s   00:02    
ruby193-syntax-1.0.0nb1.tgz              100%   88KB  87.7KB/s  87.7KB/s   00:01    
ruby193-safe_yaml-0.9.5.tgz              100%  121KB 120.6KB/s  50.6KB/s   00:01    
ruby193-redcarpet-3.0.0.tgz              100%  175KB 175.3KB/s 175.3KB/s   00:01    
ruby193-maruku-0.6.1.tgz                 100%  505KB 126.3KB/s 128.1KB/s   00:04    
ruby193-liquid-2.5.2.tgz                 100%  233KB 116.3KB/s 139.5KB/s   00:02    
ruby193-kramdown-0.14.2.tgz              100%  709KB 141.8KB/s 183.1KB/s   00:05    
ruby193-directory-watcher-1.5.1.tgz      100%  177KB  88.6KB/s  15.1KB/s   00:02    
ruby193-commander-4.1.5.tgz              100%  141KB 141.3KB/s 141.3KB/s   00:00    
ruby193-colorator-0.1.tgz                100%   58KB  57.6KB/s  57.6KB/s   00:01    
ruby193-classifier-1.3.3.tgz             100%  119KB 118.6KB/s 118.6KB/s   00:01    
ruby193-base-1.9.3p484.tgz               100% 8252KB 125.0KB/s 126.9KB/s   01:06    
ruby193-pygments.rb-0.5.2.tgz            100% 2860KB 238.4KB/s 406.3KB/s   00:12    
ruby193-jekyll-1.2.0.tgz                 100%  656KB  93.7KB/s  30.6KB/s   00:07    
installing packages...
installing libxcb-1.9.3...
installing libXdmcp-1.1.1...
installing libXau-1.0.8...
installing libXext-1.3.2...
installing libX11-1.6.2...
installing libffi-3.0.13...
installing gdbm-1.10nb3...
installing tk-8.5.15...
installing tcl-8.5.15...
installing ruby193-tk-1.9.3p484nb1...
===========================================================================
$NetBSD: MESSAGE,v 1.1.1.1 2011/11/08 16:10:51 taca Exp $

Note that this is a minimal package.

The original ruby-1.9.3-p484 distribution includes more extension
libraries, some of which are provided as separate packages:

    databases/ruby-gdbm:        GDBM module
    devel/ruby-curses:      Curses module
    devel/ruby-fiddle:      Fiddle module
    devel/ruby-readline:        readline module
    x11/ruby-tk:            Tk modules

===========================================================================
installing ruby193-readline-1.9.3p484...
installing ruby193-gdbm-1.9.3p484...
===========================================================================
$NetBSD: MESSAGE,v 1.2 2001/11/11 06:17:27 hubertf Exp $

This package doesn't contain any manuals nor examples, but
it wouldn't be a problem, as the interface of this module
is exactly the same as DBM and SDBM, except GDBM has a
"reorganize" method.  See the DBM module's document included
in the ruby-man package, and gdbm(3) manpage included in the
gdbm package.

===========================================================================
installing ruby193-fiddle-1.9.3p484...
installing ruby193-curses-1.9.3p484nb1...
installing libyaml-0.1.4...
installing ruby193-fast-stemmer-1.0.2...
installing ruby193-highline-1.6.19...
installing ruby193-syntax-1.0.0nb1...
installing ruby193-safe_yaml-0.9.5...
installing ruby193-redcarpet-3.0.0...
installing ruby193-maruku-0.6.1...
installing ruby193-liquid-2.5.2...
installing ruby193-kramdown-0.14.2...
installing ruby193-directory-watcher-1.5.1...
installing ruby193-commander-4.1.5...
installing ruby193-colorator-0.1...
installing ruby193-classifier-1.3.3...
installing ruby193-base-1.9.3p484...
installing ruby193-pygments.rb-0.5.2...
installing ruby193-jekyll-1.2.0...
pkg_install warnings: 0, errors: 0
reading local summary...
processing local summary...
updating database: 100%
marking ruby193-jekyll-1.2.0 as non auto-removable
```

but jekyll bailed out with:

```
$ jekyll serve
/usr/pkg/lib/ruby/1.9.3/rubygems/dependency.rb:247:in `to_specs': Could not find directory_watcher (~> 1.4.1) amongst [bigdecimal-1.1.0, classifier-1.3.3, colorator-0.1, commander-4.1.5, directory_watcher-1.5.1, fast-stemmer-1.0.2, highline-1.6.19, io-console-0.3, jekyll-1.2.0, json-1.5.5, kramdown-0.14.2, liquid-2.5.2, maruku-0.6.1, minitest-2.5.1, posix-spawn-0.3.6, pygments.rb-0.5.2, rake-0.9.2.2, rdoc-3.9.5, redcarpet-3.0.0, safe_yaml-0.9.5, syntax-1.0.0, yajl-ruby-1.1.0] (Gem::LoadError)
    from /usr/pkg/lib/ruby/1.9.3/rubygems/specification.rb:783:in `block in activate_dependencies'
    from /usr/pkg/lib/ruby/1.9.3/rubygems/specification.rb:772:in `each'
    from /usr/pkg/lib/ruby/1.9.3/rubygems/specification.rb:772:in `activate_dependencies'
    from /usr/pkg/lib/ruby/1.9.3/rubygems/specification.rb:756:in `activate'
    from /usr/pkg/lib/ruby/1.9.3/rubygems.rb:1236:in `gem'
    from /usr/pkg/bin/jekyll:22:in `<main>'
```

It looks like it's looking for directory_watcher 1.4.1, but I've got 1.5.1:

```
$ pkgin list | grep ruby.*directory
ruby193-directory-watcher-1.5.1 FAM-like facility for Ruby

$ find /usr/pkg/lib/ruby -name '*directory*'
/usr/pkg/lib/ruby/gems/1.9.3/cache/directory_watcher-1.5.1.gem
/usr/pkg/lib/ruby/gems/1.9.3/doc/directory_watcher-1.5.1
/usr/pkg/lib/ruby/gems/1.9.3/doc/directory_watcher-1.5.1/rdoc/lib/directory_watcher
/usr/pkg/lib/ruby/gems/1.9.3/doc/directory_watcher-1.5.1/rdoc/lib/directory_watcher_rb.html
/usr/pkg/lib/ruby/gems/1.9.3/doc/highline-1.6.19/ri/HighLine/Question/directory-i.ri
/usr/pkg/lib/ruby/gems/1.9.3/gems/directory_watcher-1.5.1
/usr/pkg/lib/ruby/gems/1.9.3/gems/directory_watcher-1.5.1/lib/directory_watcher
/usr/pkg/lib/ruby/gems/1.9.3/gems/directory_watcher-1.5.1/lib/directory_watcher.rb
/usr/pkg/lib/ruby/gems/1.9.3/gems/directory_watcher-1.5.1/spec/directory_watcher_spec.rb
/usr/pkg/lib/ruby/gems/1.9.3/specifications/directory_watcher-1.5.1.gemspec
```

I worked around this by installing the gem with "gem193" (which is the "gem" executable that pkgsrc installed):

```
$ sudo gem193 install directory_watcher -v 1.4.1
Fetching: directory_watcher-1.4.1.gem (100%)
Successfully installed directory_watcher-1.4.1
1 gem installed
Installing ri documentation for directory_watcher-1.4.1...
Installing RDoc documentation for directory_watcher-1.4.1...
```

and that seems to have worked:

```
$ jekyll serve
Configuration file: /Users/dap/Desktop/se411-pages/_config.yml
            Source: /Users/dap/Desktop/se411-pages
       Destination: /Users/dap/Desktop/se411-pages/_site
      Generating... done.
[2015-02-11 21:21:20] INFO  WEBrick 1.3.1
[2015-02-11 21:21:20] INFO  ruby 1.9.3 (2013-11-22) [i386-darwin10]
[2015-02-11 21:21:20] INFO  WEBrick::HTTPServer#start: pid=13151 port=4000
```

**Comment by jperkin** [comment]
Yeh thanks Dave, I use Jekyll too and noticed this recently.  It is fixed as of 904068f06d3e346261c95c70244e294343f1e628 which isn't part of 2014Q4.  I'll merge it back though and rebuild the jekyll package for OS X (I already merged that back for the 2014Q4 SmartOS sets).  I won't be able to fix this for older sets like 2013Q4 though.

In the meantime, you can also work around this by editing the jekyll gemspec to bump the directory_watcher dependency up to 1.5.1 - that works fine for me.

**Comment by davepacheco** [comment]
Thanks, and sorry for the dup!
